x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -782,6 +802,15 @@ private Expression resolveAndOrientJoinCondition(
                     return comp.swapLeftAndRight(); // Swapped orientation
                 }
             }
+            if (context.minimumVersion().onOrAfter(ESQL_LOOKUP_JOIN_FULL_TEXT_FUNCTION) == false) {
+                return new UnresolvedAttribute(
+                    condition.source(),
+                    "unsupported",
+                    "Lookup join on condition is not supported on the remote node,"
+                        + " consider upgrading the remote node. Unsupported join filter expression:"
+                        + condition.sourceText()
+                );
+            }
```

Comment:
Yep! We can fail early now, nice. Let's chat about whether we should, because we'll force users to exclude older remote clusters from using the new feature. (Rather than feeding the old clusters garbage and just failing at runtime or during serialization.)